docs/usage/21-provisioning-ingresses-with-tls.adoc

```diff
@@ -0,0 +1,196 @@
+:important-caption: :heavy_exclamation_mark:
+:caution-caption: :fire:
+:warning-caption: :warning:
+endif::[]
+
+= Provisioning Kubernetes Ingress(es) with TLS
+:icons: font
+
+[discrete]
+== Table of Contents
+toc::[]
+
+== Instructions
+
+=== Using `dklb` to provision a Kubernetes Ingress with TLS enabled
+
+Please read <<20-provisioning-ingresses.adoc#,Provisioning Kubernetes
+Ingress(es)>> first, before continuing.
+
+To expose an HTTP application running on MKE with a TLS certificate to either
+inside or outside the DC/OS cluster, a Kubernetes
+https://kubernetes.io/docs/concepts/services-networking/ingress/[`Ingress`]
+resource must be created.
+
+=== Enabling TLS
+
+The need to enable TLS for a given Ingress resource is signaled by the presence
+of a non-empty `.spec.tls` field. The presence of that field will cause dklb to
+enable HTTPS for the Ingress resource. The user is responsible for making sure
+that the specified secrets contain certificates for all hosts specified in it.
+EdgeLB will know how to pick the best certificate for a given host.
+
+=== Customizing the target EdgeLB pool
+
+You can use the same options to customize you EdgeLB pool as described
+<<20-provisioning-ingresses.adoc#,here>>.
+
+=== Customizing HTTP and HTTPS frontend bind ports
+
+By default `dklb` will use port 80 for the HTTP frontend port and 443 for the
+HTTPS frontend port. You can define custom ports via the `.frontends.http.port`
+and `.frontends.https.port` fields.
+
+[source,yaml]
+----
+kubernetes.dcos.io/dklb-config: |
+  frontends:
+    http:
+      mode: [disabled|enabled|redirect]
+      port: <frontend-http-bind-port>
+    https:
+      port: <frontend-https-bind-port>
+----
+
+The `.frontends.http.mode` field has the following semantics:
+
+[frame="topbot",options="header"]
+|================================
+| Value    | Description
+| disabled | Plain HTTP serving will be disabled altogether (i.e. the <cluster-name>:<namespace>:<name>:http EdgeLB frontend will be removed from the target EdgeLB pool).
+| enabled  | Default. Plain HTTP serving will be enabled., and traffic will be sent to the intended backends.
+| redirect | The plain HTTP frontend will respond with 307 TEMPORARY REDIRECT to all requests.
+|================================
+
+WARNING: Changing the value of this field after the `Ingress` resource is created is supported, but may cause disruption (as the target EdgeLB pool will most likely be re-deployed).
+
+== Example
+
+=== Exposing an HTTPS "echo" application
+
+Create an HTTP "echo" application and a service of type NodePort
+
+[source,console]
+----
+$ kubectl run --restart=Never --image hashicorp/http-echo --labels app=http-echo-1,owner=dklb --port 80 http-echo-1 -- -listen=:80 --text='Hello from http-echo-1!'
+$ kubectl expose pod http-echo-1 --port 80 --target-port 80 --type NodePort --name "http-echo-1"
+----
+
+Create a self-signed certificate for dummy domain foo.bar.com and store it in a
```

Comment:
Might need to provide an example with a CA-issued certificate. Some will also require intermediate certificates - is that supported?

Reply:
The other example I can take care in a follow up PR. I pinged Deepak regarding the intermediate certificate question to understand if EdgeLB has any limitation.

Reply:
Intermediate certs will be needed for clients to properly verify server cert chain.